src/store/actions/posting-project-user/profile/CreatePageAction.js

```diff
@@ -0,0 +1,117 @@
+// Firebase database
+import {firebase_db} from '../../../../firebase-config';
+import { firebase_storage } from "../../../../firebase-config";
+import { storage } from "firebase";
+
+// Constants
+import { testing_organization_id} from '../../../../constants/testing-keys';
+
+
+// Export all the action types
+export const CREATE_PROFILE = "CREATE_PROFILE";
+
+/******************************** ACTIONS ********************************/
+
+  
+      /**
+   * @summary Handle upload the image to storage of Firebase
+   * @param {Object} imageFile An object containning such as {name:...,type:img/png,...vv}
+   * @return {void}
+   * @author TrNgTien
+   */
+const handleUpLoad = async (imageFile)=>{
+
+    // let imageUrl;
+   
+    const uploadTask = firebase_storage.ref(`images/${imageFile.name}`)
+
+    return new Promise(async (res, rej) => {
+        uploadTask.put(imageFile).on(
+            "state_changed",
+            snapshot =>{},
+            error =>{
+                console.log("big erroor",error);
+                rej(error)
+            },
+            ()=>{ 
+                firebase_storage
+                .ref("images")
+                .child(imageFile.name)
+                .getDownloadURL()
+                .then(url=>{
+                    // setUrl(url);
+                    console.log(url)
+                    res(url)
+                    });
+            }
+        );
+    })
+    
+        // return imageUrl ? imageUrl : "";
+        // ternary operator
+
+}
+ 
+
+
+ 
+
+
+
+export const _createProfile_ppu = (orgId, orgName, description, location, University,email,phoneNumber,facebook,imageFile) => {
+
+
+    
+
+    /**
+     * @summary Asynchronous function calling the database to push the data
+     * @param {function} dispatch => Function used to send the action type and data to the Redux reducer
+     * @param {function} getState => Function used to access the current state of the application
+     * @returns {void}
+     * @author TrNgTien
+     */
+    return async (dispatch, getState) => {
+     
+        const imageUrl = await handleUpLoad(imageFile);
+
+        console.log('url in createProfile', imageUrl)
+
+        try {
+            await
+            firebase_db
+            .collection('organization')
+            .doc(`${testing_organization_id}`)
+            .set({
```

Comment:
`set` method is used for changing the document. We'll accept the code in the situation where deadline is a priority.

--> Notice that, the task is to **add** (aka create a **new document**). Please take time research on this _(keyword: **add** method of firebase document)_

Reply:
From [Firebase doc](https://firebase.google.com/docs/firestore/manage-data/add-data) 

> Behind the scenes, .add(...) and .doc().set(...) are completely equivalent, so you can use whichever is more convenient.

Reply:
From[StackOverFlow](https://stackoverflow.com/questions/47474522/firestore-difference-between-set-and-add#:~:text=When%20you%20use%20set(),identified%20by%20some%20unique%20id.&text=When%20you%20use%20add(),unique%20id%20assigned%20to%20it.)